src/overlays/actors/ovl_Bg_Haka_Bombwall/z_bg_haka_bombwall.c

```diff
@@ -98,53 +99,54 @@ void func_80BD5E6C(BgHakaBombwall* this, PlayState* play) {
         if (temp_fs0 > 75.0f) {
             temp_fs0 -= 150.0f;
         }
-        spC8.x = temp_fs0;
 
+        pos.x = temp_fs0;
         phi_fs1 += 5;
-        spC8.y = phi_fs1;
+        pos.y = phi_fs1;
+        pos.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
 
-        spC8.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
+        vel.x = ((Rand_ZeroOne() - 0.5f) * 5.0f) + (temp_fs0 * (4.0f / 75.0f));
+        vel.y = (Rand_ZeroOne() * 7.0f) - 2.0f;
+        vel.z = (Rand_ZeroOne() * 4.0f) - 2.0f;
 
-        spBC.x = ((Rand_ZeroOne() - 0.5f) * 5.0f) + (temp_fs0 * (4.0f / 75.0f));
-        spBC.y = (Rand_ZeroOne() * 7.0f) - 2.0f;
-        spBC.z = (Rand_ZeroOne() * 4.0f) - 2.0f;
+        Matrix_MultVec3f(&pos, &mtxPos);
+        Matrix_MultVec3f(&vel, &mtxVel);
 
-        Matrix_MultVec3f(&spC8, &spE0);
-        Matrix_MultVec3f(&spBC, &spD4);
+        mtxPos.x += this->dyna.actor.world.pos.x;
+        mtxPos.y += this->dyna.actor.world.pos.y;
+        mtxPos.z += this->dyna.actor.world.pos.z;
 
-        spE0.x += this->dyna.actor.world.pos.x;
-        spE0.y += this->dyna.actor.world.pos.y;
-        spE0.z += this->dyna.actor.world.pos.z;
+        if (1) {}
 
         if ((i & 3) == 0) {
             phi_s0 = 32;
-            func_800BBFB0(play, &spE0, 60.0f, 2, 100, 120, 1);
+            func_800BBFB0(play, &mtxPos, 60.0f, 2, 100, 120, 1);
         } else {
             phi_s0 = 64;
         }
 
-        if (i % 2 != 0) {
+        if ((i % 2) != 0) {
             phi_s0 |= 1;
             phi_t0 = 1;
         } else {
             phi_t0 = 0;
         }
 
-        if (D_80BD64CC[i & 3] >= 16) {
+        scale = rockScale[i & 3];
```

Comment:
The question I always ask, can this be a `% ARRAY_COUNT`?
```suggestion
        scale = sRockScale[i % ARRAY_COUNT(sRockScale)];
```

Reply:
It isn't matching this way for me, if you can find a way (using the scratch above) to make that happen it would be great!